docs/src/main/tut/datatypes/validated.md

```diff
@@ -12,7 +12,351 @@ Response comes back saying your username can't have dashes in it, so you make so
+
+### An iteration with `Validated`
+
+Time to do some refactoring! We're going to try a `Validated` approach:
+
+```tut:silent
+import cats.data._
+import cats.data.Validated._
+import cats.implicits._
+
+def validateUserName(userName: String): Validated[DomainValidation, String] = {
+  if (userName.matches("^[a-zA-Z0-9]+$")) Valid(userName) else Invalid(UsernameHasSpecialCharacters)
+}
+
+def validatePassword(password: String): Validated[DomainValidation, String] = {
+  if (password.matches("(?=^.{10,}$)((?=.*\\d)|(?=.*\\W+))(?![.\\n])(?=.*[A-Z])(?=.*[a-z]).*$")) Valid(password)
+  else Invalid(PasswordDoesNotMeetCriteria)
+}
+
+def validateFirstName(firstName: String): Validated[DomainValidation, String] = {
+  if (firstName.matches("^[a-zA-Z]+$")) Valid(firstName) else Invalid(FirstNameHasSpecialCharacters)
+}
+
+def validateLastName(lastName: String): Validated[DomainValidation, String] = {
+  if (lastName.matches("^[a-zA-Z]+$")) Valid(lastName) else Invalid(LastNameHasSpecialCharacters)
+}
+
+def validateAge(age: Int): Validated[DomainValidation, Int] = {
+  if (age >= 18 && age <= 75) Valid(age) else Invalid(AgeIsInvalid)
+}
+```
+```tut:book:fail
+def validateForm(username: String, password: String, firstName: String, lastName: String, age: Int): Validated[DomainValidation, RegistrationData] = {
+  for {
+    validatedUserName <- validateUserName(username)
+    validatedPassword <- validatePassword(password)
+    validatedFirstName <- validateFirstName(firstName)
+    validatedLastName <- validateLastName(lastName)
+    validatedAge <- validateAge(age)
+  }
+    yield RegistrationData(validatedUserName, validatedPassword, validatedFirstName, validatedLastName, validatedAge)
+}
+```
+
+Looks similar to the first version. What we've done here was to use `Validated` instead of `Either`. Please note that our `Right` is now a `Valid` and `Left` is an `Invalid`.
+Remember, our goal is to get all the validation errors for displaying it to the user.
+
+But this approach won't compile, as you can see in the previous snippet. Why?
+
+Without diving into details about monads, a for-comprehension uses the `flatMap` method for composition. Monads like `Either` can be composed in that way, but the thing with `Validated` is that it isn't a monad, but an [_Applicative Functor_](../typeclasses/applicativetraverse.html).
+That's why you see the message: `error: value flatMap is not a member of cats.data.Validated[DomainValidation,String]`.
+
+So, how do we do here?
+
+### Meeting applicative
+
+We have to look into another direction: a for-comprehension plays well in a fail-fast scenario, but the structure in our previous example was designed to catch one error at a time, so, our next step is to tweak the implementation a bit.
+
+```tut:silent
+import cats.data._
+import cats.data.Validated._
+import cats.implicits._
+
+sealed trait FormValidatorNel {
+
+  type ValidationResult[A] = Validated[NonEmptyList[DomainValidation], A]
```

Comment:
Since you are using `validNel` and `invalidNel` below, I think you can use `ValidatedNel` here as well.